<launch>

  <!-- This is actually derived from freenect_launch/freenect-registered-xyzlaunch-->
  <include file="$(find freenect_launch)/launch/freenect.launch">

    <!-- use device registration -->
    <arg name="depth_registration"              value="true" /> 

    <arg name="rgb_processing"                  value="true" /> 
    <arg name="ir_processing"                   value="false" />

    <arg name="depth_processing"                value="true" />
    <arg name="depth_registered_processing"     value="true" />
    <arg name="disparity_processing"            value="true" />
    <arg name="disparity_registered_processing" value="true" />

    <arg name="sw_registered_processing"        value="false" />
    <arg name="hw_registered_processing"        value="true" />

  </include>

  <node pkg="test_kinect" type="testDisparity.py" name="testDisparity">
    <remap from="image_out" to="/camera/filtered"/>
    <remap from="disparity_in" to="/camera/depth_registered/disparity"/>
    <remap from="rgb_in" to="/camera/rgb/image_raw"/>
  </node>

  <node pkg="image_view" type="disparity_view" name="dispview">
    <remap from="/image" to="/camera/depth_registered/disparity"/>
  </node>
  <node pkg="image_view" type="image_view" name="imgview">
    <remap from="/image" to="/camera/filtered"/>
  </node>

</launch>
